tools/gradle/commons/integrations/image.gradle

```diff
@@ -21,3 +21,5 @@ task buildImage(type: DockerBuildImage) {
     // Docker will still cache the artifact, so it shouldn't be much slower.
     outputs.upToDateWhen { false }
 }
+
+buildImage.dependsOn(build)
```

Comment:
fixes the thing that was driving us crazy in the java integrations where buildImage did not depend on build happening first.